src/openassetio-python/cmodule/src/hostApi/ManagerBinding.cpp

```diff
@@ -74,50 +74,60 @@ void registerManager(const py::module& mod) {
   pyManager
       .def(py::init(RetainCommonPyArgs::forFn<&Manager::make>()),
            py::arg("managerInterface").none(false), py::arg("hostSession").none(false))
-      .def("identifier", &Manager::identifier)
-      .def("displayName", &Manager::displayName)
-      .def("info", &Manager::info)
-      .def("settings", &Manager::settings)
-      .def("initialize", &Manager::initialize, py::arg("managerSettings"))
-      .def("flushCaches", &Manager::flushCaches)
+      .def("identifier", &Manager::identifier, py::call_guard<py::gil_scoped_release>{})
```

Comment:
Genius, will look into it

Reply:
Great idea - capturing our off-line discussion, things to consider whilst doing this:
- Important to make it clear that this isn't vanilla pybind, as use a lot of that in the surrounding context, and the instantiation is generally off-screen, so it may not be apparent that this is a magic `def`, so people could easily re-add the release thinking its not there.
- We want to encourage people to think carefully about the GIL here (similarly to `.none(false)`), so not sure if 'release by default' is always the thing to do, especially for the more trivial and performance critical bound code.

Reply:
Did you make a call on this? Am fine just binning it, there's arguments to the contray.

Reply:
As discussed, it's a neat idea, but after prototyping we can see that gets a bit messy when you have files with a mixture of `py::class_` and this new class, and are using a mixture of methods.  

E.g. `def`, `def_static`, `def_readonly_static`, etc, some of which must be replaced with GIL/no-GIL variants, and some which should retain previous behaviour. All methods must be overridden in the subclass too, in order for method chaining (i.e. `.def(...).def(...)`) to work.

On balance it feels like the maintenance on onboarding overhead, along with the messiness in some bindings (e.g. `ManagerFactory`) tips the scales against the benefit of additional explicitness and neater signatures.

Reply:
Thanks @feltech as @elliotcmorris is out till tomorrow lunch, and this is the only remaining thread - after our off-line huddle, I don't think hell mind us getting this ready for merge.